docs/latest/design/extending-envoy-gateway.md

```diff
@@ -144,6 +144,62 @@ latency between Envoy and the authentication service and also allows the end use
 
 ![Control Plane Extension Example](../images/extension-auth-example.png)
 
+## Design Decisions
+
+- Since an HTTPRoute resource corresponds to an [Envoy Proxy Route resource][], the name of the Route is set using the name and namespace of the HTTPRoute that generated it so that any extension looking at the
+xDS resources provided by Envoy Gateway can have the same context about which Routes to modify.
+- The responsibility of managing scale and performance of hook calls and modification of xDS resources falls on the extension controller, not Envoy Gateway
+
+## Example: Post xDS Hook API
+
+```go
+// TODO: If Envoy Gateway is modified to watch resources introduced by extensions, then this API would be updated to send those resources over to the extension
+// for any hook calls.
+type PostHTTPListenerTranslationRequest struct {
+    state         protoimpl.MessageState
+    sizeCache     protoimpl.SizeCache
+    unknownFields protoimpl.UnknownFields
```

Comment:
you can omit the private fields since they are just generated proto struct boilerplate